packages/content-management/table_list_view_table/src/components/user_filter_panel.tsx

```diff
@@ -0,0 +1,177 @@
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { FC } from 'react';
+import React from 'react';
+import { EuiFilterButton, EuiIconTip, useEuiTheme } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { UserProfile, UserProfilesPopover } from '@kbn/user-profile-components';
+import { useQuery, QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { i18n } from '@kbn/i18n';
+
+interface Context {
+  onSelectedUsersChange: (users: string[]) => void;
+  selectedUsers: string[];
+  allUsers: string[];
+  bulkGetUserProfiles?: (uids: string[]) => Promise<UserProfile[]>;
+}
+
+const UserFilterContext = React.createContext<Context | null>(null);
+const queryClient = new QueryClient({
+  defaultOptions: { queries: { retry: false, staleTime: 30 * 60 * 1000 } },
+});
+
+export const UserFilterContextProvider: FC<Context> = ({ children, ...props }) => {
+  if (!props.bulkGetUserProfiles) {
+    return <>{children}</>;
+  }
+
+  return (
+    <QueryClientProvider client={queryClient}>
+      <UserFilterContext.Provider value={props}>{children}</UserFilterContext.Provider>
+    </QueryClientProvider>
+  );
+};
+
+export const NULL_USER = 'no-user';
+
+export const UserFilterPanel: FC<{}> = () => {
+  const { euiTheme } = useEuiTheme();
+  const componentContext = React.useContext(UserFilterContext);
+  if (!componentContext)
+    throw new Error('UserFilterPanel must be used within a UserFilterContextProvider');
+  if (!componentContext.bulkGetUserProfiles)
+    throw new Error('UserFilterPanel must be used with a bulkGetUserProfiles function');
+
+  const { onSelectedUsersChange, selectedUsers } = componentContext;
+
+  const [isPopoverOpen, setPopoverOpen] = React.useState(false);
+  const [searchTerm, setSearchTerm] = React.useState('');
+
+  const query = useQuery({
+    queryKey: ['user-filter-suggestions', componentContext.allUsers],
+    queryFn: () => componentContext.bulkGetUserProfiles!(componentContext.allUsers),
+    enabled: isPopoverOpen,
+  });
+
+  const usersMap = React.useMemo(() => {
+    if (!query.data) return {};
+    return query.data.reduce((acc, user) => {
+      acc[user.uid] = user;
+      return acc;
+    }, {} as Record<string, UserProfile>);
+  }, [query.data]);
+
+  const visibleOptions = React.useMemo(() => {
+    if (!query.data || query.data.length === 0) return [];
+    // attach null to the end of the list to represent the "no creator" option
+    const users = [...query.data, null];
+
+    if (!searchTerm) {
+      return users;
+    }
+
+    return users.filter((user) => {
+      // keep the "no creator" option if it's selected
+      if (!user) {
+        return selectedUsers.includes(NULL_USER);
+      }
+
+      // keep the user if it's selected
+      if (selectedUsers.includes(user.uid)) return true;
```

Comment:
yes, I think it is better ux.

Reply:
I think it is a better ux. We want to show the selected users while searching even if they don't match, so that the user can unselect without clearing the filter first. 

Basically, selected should always stay in the list